diff --git a/app/pages/weather.component.ts b/app/pages/weather.component.ts
index c61e5ec..b0393b8 100644
--- a/app/pages/weather.component.ts
+++ b/app/pages/weather.component.ts
@@ -9,32 +9,64 @@ import { CurrentConditions } from '../types/Weather';
 export class WeatherComponent implements OnInit {

   currentConditions: CurrentConditions[];
+  currentConditionsResolution: CurrentConditions[];

-  constructor() { }
+  constructor() {
+    this.currentConditionsResolution = [{
+      "LocalObservationDateTime": "2017-09-20T05:45:00+02:00",
+      "EpochTime": 1505879100,
+      "WeatherText": "Some clouds",
+      "WeatherIcon": 36,
+      "IsDayTime": false,
+      "Temperature": {
+        "Metric": {
+          "Value": 16.4,
+          "Unit": "C",
+          "UnitType": 17
+        },
+        "Imperial": {
+          "Value": 62,
+          "Unit": "F",
+          "UnitType": 18
+        }
+      },
+      "MobileLink": "http://m.accuweather.com/en/al/fushe-verri/1557905/current-weather/1557905?lang=en-us",
+      "Link": "http://www.accuweather.com/en/al/fushe-verri/1557905/current-weather/1557905?lang=en-us"
+    }];
+  }

   ngOnInit(): void {
-    this.currentConditions = [
-      {
-        "LocalObservationDateTime": "2017-09-20T05:45:00+02:00",
-        "EpochTime": 1505879100,
-        "WeatherText": "Some clouds",
-        "WeatherIcon": 36,
-        "IsDayTime": false,
-        "Temperature": {
-          "Metric": {
-            "Value": 16.4,
-            "Unit": "C",
-            "UnitType": 17
-          },
-          "Imperial": {
-            "Value": 62,
-            "Unit": "F",
-            "UnitType": 18
-          }
-        },
-        "MobileLink": "http://m.accuweather.com/en/al/fushe-verri/1557905/current-weather/1557905?lang=en-us",
-        "Link": "http://www.accuweather.com/en/al/fushe-verri/1557905/current-weather/1557905?lang=en-us"
-      }
-    ]
+    this.allWrapper();
+  }
+
+  private getCurrentLocation(): Promise<any> {
+    return Promise.resolve([40,70]);
+  }
+
+  private getCurrentWeather(location: any): Promise<CurrentConditions[]> {
+    return Promise.resolve(this.currentConditionsResolution);
+  }
+
+  private getForecastWeather(location: any): Promise<any> {
+    return Promise.resolve(this.currentConditionsResolution);
   }
+
+  private async allWrapper(): Promise<void> {
+    try {
+
+      const curLoc = await this.getCurrentLocation();
+
+      this.getCurrentWeather(curLoc).then(
+        (weather) => {let something = weather}
+      );
+      this.getForecastWeather(curLoc).then(
+        (weather) => {let something = weather}
+      );
+
+    } catch(e) {
+
+    }
+
+  }
+
 }
\ No newline at end of file